internal/http/http.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2020 Red Hat, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"k8s.io/klog"
+)
+
+const (
+	metricsPath = "/metrics"
+)
+
+// ServeExporterMetrics serves exporter self metrics on metricsPath.
+func ServeExporterMetrics(host string, port int, registry *prometheus.Registry) {
```

Comment:
These two functions in this pkg look very similar and do a lot of the same, maybe you can refactor them a bit.